packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts

```diff
@@ -0,0 +1,91 @@
+import winston from "winston";
+
+import { createEtherscanLinkMarkdown, createFormatFunction, PublicNetworks } from "@uma/common";
+import { getAbi } from "@uma/contracts-node";
+import { InsuredBridgeL1Client, InsuredBridgeL2Client, GasEstimator } from "@uma/financial-templates-lib";
+
+export enum RelaySubmitType {
+  Slow,
+  SpeedUp,
+  Instant,
+  Ignore,
+}
+
+export class CrossDomainFinalizer {
+  constructor(
+    readonly logger: winston.Logger,
+    readonly gasEstimator: GasEstimator,
+    readonly l1Client: InsuredBridgeL1Client,
+    readonly l2Client: InsuredBridgeL2Client,
+    readonly account: string
+  ) {}
+  async checkForBridgeableL2TokensAndBridge() {
+    this.logger.debug({ at: "AcrossRelayer#CrossDomainFinalizer", message: "Checking bridgeable L2 tokens" });
+
+    const whitelistedTokenEvents = await this.l1Client.bridgeAdmin.getPastEvents("WhitelistToken", {
```

Comment:
Although its inconsistent to query events in bots in this repo, I think this is a performance optimization as we don't want to query `WhitelistToken` in the InsuredBridgeL1Client

Reply:
correct. it's also way less code. I could move it into a helper perhaps, like that of the index helpers.